app/frontend/src/features/messages/groupchats/groupChatsActions.ts

```diff
@@ -0,0 +1,74 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import { User } from "../../../pb/api_pb";
+import { GroupChat } from "../../../pb/conversations_pb";
+import { RootState } from "../../../reducers";
+import { service } from "../../../service";
+import { groupChatsFetched, messagesFetched, groupChatSet } from ".";
+
+type FetchGroupChatsArguments = void;
+
+export const fetchGroupChats = createAsyncThunk<
+  void,
+  FetchGroupChatsArguments,
+  { state: RootState }
+>("hostRequests/fetchGroupChats", async (_, thunkApi) => {
+  const groupChats = await service.conversations.listGroupChats();
+  thunkApi.dispatch(groupChatsFetched(groupChats));
+});
```

Comment:
This is what I meant when I said it would be better to get rid of setMessages. I missed this last night must've been looking at the wrong diff, sorry!